packages/block-editor/src/components/inserter/search-items.js

```diff
@@ -54,28 +54,42 @@ export const searchBlockItems = (
 		return items;
 	}
 
-	return searchItems( items, searchTerm, {
+	const config = {
 		getCategory: ( item ) =>
 			find( categories, { slug: item.category } )?.title,
 		getCollection: ( item ) =>
 			collections[ item.name.split( '/' )[ 0 ] ]?.title,
-		getVariations: ( item ) =>
-			( item.variations || [] ).map( ( variation ) => variation.title ),
-	} ).map( ( item ) => {
+		getVariations: ( { variations = [] } ) =>
+			Array.from(
+				variations.reduce(
+					( accumulator, { title, keywords = [] } ) => {
+						accumulator.add( title );
+						keywords.forEach( ( keyword ) =>
+							accumulator.add( keyword )
+						);
```

Comment:
Do you think the code here will be simpler if use arrays instead of Set? 

`return [ ...accumulator, title, ...keywords ]` (and not more Arrary.from)

Reply:
Yes it would but variations can have same keywords and will be added to the array duplicates. This became to me really obvious with `embed` sharing many common keywords.